rpc/jsonrpc_state_server.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package rpc
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ava-labs/avalanchego/trace"
+)
+
+type stateReader interface {
+	Tracer() trace.Tracer
```

Comment:
It seems weird to have the tracer on the interface for this. I think it would make more sense to just make this an implementation detail and pass the tracer through `NewJsonRPCStateServer` when it's constructed.

Reply:
This is the current style, so I think this is good for this PR and we can separately refactor to avoid this pattern if desired